### PR TITLE
update to show crypto ikev2 sa detail

### PR DESCRIPTION
**Pull request by ciesinsn**
## Description
Updated IOSXE show crypto ikev2 sa detail to parse output where remote id is not a IP address

## Motivation and Context
Related to issue https://github.com/CiscoTestAutomation/genieparser/issues/718

Current regex did not parse remote id that wasn't a IP address

## Impact (If any)
None

## Screenshots:
Screen Shot 2022-12-23 at 12 11 40 AM
Screen Shot 2022-12-23 at 12 12 30 AM
Screen Shot 2022-12-23 at 12 12 44 AM
Screen Shot 2022-12-23 at 12 12 54 AM

## Checklist:

- [x] I have updated the changelog.
- [ ] I have updated the documentation (If applicable).
- [x] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [x] All new code passed compilation.

### CODE REVIEW

**Comment by ciesinsn** [comment]
Accidentally pushed another commit with this one in the PR.  I since removed it.

File: src/genie/libs/parser/iosxe/tests/test_show_crypto_ikev2_sa_detail.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
+# Import the Python mock functionality
```
Comment (Taarini) [inline src/genie/libs/parser/iosxe/tests/test_show_crypto_ikev2_sa_detail.py:1]: This file can be removed.

**Review by Taarini — CHANGES_REQUESTED** [review]

File: src/genie/libs/parser/iosxe/tests/test_show_crypto_ikev2_sa_detail.py (hunk 1 above)
Comment (ciesinsn) [reply src/genie/libs/parser/iosxe/tests/test_show_crypto_ikev2_sa_detail.py:1]: I can do that.  Are these files never needed in a PR?  The docs seemed to seem like they where but I got confused with the docs also saying "the old way"

**Comment by Taarini** [comment]
Hi

Thanks for your contribution. Can you please resolve conflicts.

**Review by GerriorL — APPROVED** [review]

**Pull request merged**